python-package/lightgbm/basic.py

```diff
@@ -1578,7 +1578,7 @@ def dump_text(self, filename):
 class Booster(object):
     """Booster in LightGBM."""
 
-    def __init__(self, params=None, train_set=None, model_file=None, silent=False):
+    def __init__(self, params=None, train_set=None, model_file=None, model_str=None, silent=False):
```

Comment:
Yeah, I know that for the backward compatibility new param should be the last one. But it's too illogical to split `train_set=None, model_file=None, model_str=None` params (especially `model_file=None, model_str=None`), in which at least one should be specified by a user.